### PR TITLE
Add migration system 

### PR DESCRIPTION
**Pull request by vibby**
The idea is to be able to make changes in database without starting over, especially for production

### CODE REVIEW

File: application/composer.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
-{
+s{
```
Comment (akjackson1) [inline application/composer.json]: Typo?

Comment (vibby) [reply application/composer.json]: Just a ``ctrl-S`` without a ``ctrl`` ;) fixed

**Comment by akjackson1** [comment]
I'm not familiar with this specific migration system, does it not use a schema.xml or something similar so you you don't have to run all the migrations when someone installs fresh? For example, 2 years from now when there at 50 migrations, do each of those have to run in order to install the latest schema?

**Comment by vibby** [comment]
The idea is you can have as many migrations as you want.
The first one create all the base stuff.
When you add new features, you can evolve your database, for example adding a new col in a table.
You cannot just add it to the ``schema.sql``, since it may have been executed on production already (or else where).
So you have to write the SQL code that apply the modification. And as you never know what can happen, you write the SQL code get up the migration, and to get down. 
The migrations passed are stocked in database so he never do a migration twice. That is why a migration must never change once it is in master. But another migration can modify the data model another way after.
On a fresh install, you will run the 50 migrations if they are so much, but that does not matters actually. What is important : you have a situation closest as possible from production.

Is that clear ?

**Comment by akjackson1** [comment]
@vibby I'm definitely familiar with how migrations work, but on the systems I've worked with it keeps a an up-to-date XML file with all the latest schema in there which is used for fresh installs. It also functions as a reference and changes can easily be tracked. If someone adds a new column, for example, they add the migration and update the schema file. Does this system allow for something like that?

**Comment by vibby** [comment]
I do not see the need of a separate file (xml or other). Responsibility is shared that way. The file could become different from the migrations. Having just one workflow for db state seems a better idea to me, to avoid differences.

**Pull request merged**